Add documentation on how NGINX Agent works and Install instructions (#74)
nginxinc/nginx-amplify-doc: content/faq/general.md

All communications between the agent and the backend are done securely over SSL/TLS. The agent always initiates all traffic. The backend system doesn't set up any connections back to the agent.

### Is the NGINX Agent Traffic Protected?

> All communications between the agent and the backend are done over gRPC connections.

Reviewer: Rather than state `backend` maybe we ought to use NGINX Amplify services? Sounds more "professional."

-> `All communications between the agent and the NGINX Amplify services`

> All communications between the agent and the NGINX Amplify services are done over gRPC connections.

Reviewer 2: ```suggestion
The NGINX Agent and NGINX Amplify services communicate using gRPC connections.
```

### Is the NGINX Amplify Agent Code Publicly Available?

NGINX Amplify Agent is an open-source application. It is licensed under the [2-clause BSD license](https://github.com/nginxinc/nginx-amplify-agent/blob/master/LICENSE), and the code is available in [NGINX Amplify's GitHub repository](https://github.com/nginxinc/nginx-amplify-agent).

### Is the NGINX Agent Code Publicly Available?

> NGINX Amplify Agent is an open-source application. It is licensed under the [Apache License 2.0](https://github.com/nginx/agent/blob/main/LICENSE), and the code is available in [NGINX Agent's GitHub repository](https://github.com/nginx/agent).

Reviewer 2: ```suggestion
NGINX Amplify Agent is an open-source application licensed under the [Apache License 2.0](https://github.com/nginx/agent/blob/main/LICENSE). You can find the code in the [NGINX Agent's GitHub repository](https://github.com/nginx/agent).
```
